internal/provider/provider.go

```diff
@@ -103,7 +104,11 @@ func (p *AlzProvider) Configure(ctx context.Context, req provider.ConfigureReque
 	data.SetOpinionatedDefaults()
 	configureDefaults(ctx, &data)
 
-	authOptions := data.AuthOption(azcore.ClientOptions{})
+	authOptions := data.AuthOption(azcore.ClientOptions{
+		Retry: policy.RetryOptions{
+			MaxRetries: math.MaxInt16,
```

Comment:
Any reason this is a 16 bit Int as opposed to 32? I see the data type is 32 bit: https://github.com/Azure/azure-sdk-for-go/blob/e8b2c690dd862686a39bd0f0147126fc6617016a/sdk/azcore/policy/policy.go#L96

Reply:
I was copying azapi, other than that no reason.

32k retries should be fine, right? 😐

Reply:
I'm not sure, you might need 2.1M retries. :)

Reply:
This was the source - https://github.com/Azure/terraform-provider-azapi/pull/920/files#diff-b54ea12f572de62c66bd6e2d482d80f19dc679ac1c65da72e11b64166c96dc6d